/* Copyright 2012 MaidSafe.net limited

This MaidSafe Software is licensed under the MaidSafe.net Commercial License, version 1.0 or later,
and The General Public License (GPL), version 3. By contributing code to this project You agree to
the terms laid out in the MaidSafe Contributor Agreement, version 1.0, found in the root directory
of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also available at:

http://www.novinet.com/license

Unless required by applicable law or agreed to in writing, software distributed under the License is
distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
implied. See the License for the specific language governing permissions and limitations under the
License.
*/

package maidsafe.vault.protobuf;

message MaidManagerValue {
  required int64 total_cost = 1;
  required int32 count = 2;
}

message MaidManagerMetadata {
  message PmidTotal {
    required bytes serialised_pmid_registration = 1;
    required bytes serialised_pmid_metadata = 2;  // this is a serialised PmidManagerMetadata
  }
  repeated PmidTotal pmid_totals = 1;
  required int64 total_put_data = 2;
}

// Used as data field in MM's Reply to Put request.
message Cost {
  required int32 cost = 1;
}
